src/main/java/com/github/firmwehr/gentle/firm/construction/FirmGraphBuilder.java

```diff
@@ -608,19 +609,21 @@ private Node processNewArray(Context context, SNewArrayExpression expr) {
 
 		Node memberCount = construction.newConv(processValueExpression(context, expr.size()), Mode.getLu());
 
-		return allocateMemory(construction, typeSize, memberCount, expr);
+		return allocateMemory(context, construction, typeSize, memberCount, expr);
 	}
 
 	private Node processNewObject(Context context, SNewObjectExpression expr) {
 		Construction construction = context.construction();
 		ClassType type = typeHelper.getClassType(expr.classDecl());
 
-		return allocateMemory(construction, type.getSize(), construction.newConst(1, Mode.getLu()), expr);
+		return allocateMemory(context, construction, type.getSize(), construction.newConst(1, Mode.getLu()), expr);
 	}
 
-	private Node allocateMemory(Construction construction, int typeSize, Node memberCount, SExpression source) {
+	private Node allocateMemory(
+		Context context, Construction construction, int typeSize, Node memberCount, SExpression source
```

Comment:
You can remove the construction parameter and get it from the context in the method